core/modules/startup/keyboard.js

```diff
@@ -0,0 +1,43 @@
+/*\
+title: $:/core/modules/startup/keyboard.js
+type: application/javascript
+module-type: startup
+
+Keyboard shortcut handling
+
+\*/
+(function(){
+
+/*jslint node: true, browser: true */
+/*global $tw: false */
+"use strict";
+
+// Export name and synchronous status
+exports.name = "global-keyboard";
+exports.platforms = ["browser"];
+exports.after = ["startup"];
+exports.synchronous = true;
+
+exports.startup = function() {
+	if($tw.browser) {
+		$tw.utils.each($tw.wiki.filterTiddlers("[all[shadows+tiddlers]tag[$:/tags/KeyboardShortcut]!has[draft.of]]"),function(title) {
+			var self = this;
+			var key = $tw.wiki.getTiddler(title).fields["key"];
+			$tw.wiki.addEventListener("change",function(changes) {
+				if($tw.utils.hop(changes,title)) {
+					key = $tw.wiki.getTiddler(title).fields["key"];
+				}
+			});
+
+			document.addEventListener("keydown",function(event) {
+				if(key !== undefined && $tw.keyboardManager.checkKeyDescriptors(event,$tw.keyboardManager.parseKeyDescriptors(key))) {
+					$tw.rootWidget.invokeActionString('<$navigator story="$:/StoryList" history="$:/HistoryList">' + $tw.wiki.getTiddlerText(title) + '</$navigator>',$tw.rootWidget);
```

Comment:
I'm against embedding the navigator widget like this (I meant that the workaround **for users** was to use the navigator widget around any action widgets that need it). Otherwise we're further driving the specific singular default story river into the core.